WIP: feat: add "check" command

ruda opened the pull request:
Sanity check Quipucords setup and configurations.
This command will check and print the status of important files and directories that the installer creates for running Quipucords.

Example of status: OK, Not owned by you and Missing. The return code for the command is the number of problematic files/dirs.

Output:

```
❯ ./bin/quipucords-installer check
Checking Quipucords setup and configurations...
/home/rmoura/.local/share/quipucords: OK
/home/rmoura/.local/share/quipucords/certs: OK
/home/rmoura/.local/share/quipucords/data: OK
/home/rmoura/.local/share/quipucords/db: OK
/home/rmoura/.local/share/quipucords/log: OK
/home/rmoura/.local/share/quipucords/sshkeys: OK
/home/rmoura/.config/quipucords/env: OK
/home/rmoura/.config/quipucords/env/env-ansible.env: OK
/home/rmoura/.config/quipucords/env/env-app.env: OK
/home/rmoura/.config/quipucords/env/env-celery-worker.env: OK
/home/rmoura/.config/quipucords/env/env-db.env: OK
/home/rmoura/.config/quipucords/env/env-redis.env: OK
/home/rmoura/.config/quipucords/env/env-server.env: OK
/home/rmoura/.config/containers/systemd: OK
/home/rmoura/.config/containers/systemd/quipucords-app.container: OK
/home/rmoura/.config/containers/systemd/quipucords-celery-worker.container: OK
/home/rmoura/.config/containers/systemd/quipucords-db.container: OK
/home/rmoura/.config/containers/systemd/quipucords-redis.container: OK
/home/rmoura/.config/containers/systemd/quipucords-server.container: OK
/home/rmoura/.config/containers/systemd/quipucords.network: OK

```

Relates to JIRA: DISCOVERY-729

infinitewarp reviewed (changes requested):
The title of the original feature request was `Ensure appropriate ownership and permissions of required host directories` with this acceptance criteria:

> If a ~/.local (or other required directory or file) has bad ownership or mode, when running the installer's "check" command, the installer detects the problem, reports an error, and exits with a non-zero status. 

However, it looks like this PR only checks the existence and ownership of the directories and files, not the permissions (modes) of them.

I pulled and switched to your branch, and I "broke" some file and directory modes like this on my local VM:

```
chmod 000 /home/brasmith/.local/share/quipucords/certs/server.key
chmod 000 /home/brasmith/.local/share/quipucords/db/userdata
```

Confirming that they are indeed "broken":

```
$ ls -la /home/brasmith/.local/share/quipucords/certs/server.key
----------. 1 brasmith 100000 1704 Dec  2 17:12 /home/brasmith/.local/share/quipucords/certs/server.key
$ cat /home/brasmith/.local/share/quipucords/certs/server.key
cat: /home/brasmith/.local/share/quipucords/certs/server.key: Permission denied
$ ls -la /home/brasmith/.local/share/quipucords/db/userdata
ls: cannot open directory '/home/brasmith/.local/share/quipucords/db/userdata': Permission denied
```

When I run the new `check` command and search for those paths in the output, they appear to be "OK" when that is incorrect, and the script also exits with a successful `0` exit code which is also incorrect:

```
$ ./bin/quipucords-installer check | grep 'server.key\|userdata'
/home/brasmith/.local/share/quipucords/certs/server.key: OK
/home/brasmith/.local/share/quipucords/db/userdata: OK
```

```
$ ./bin/quipucords-installer check &> /dev/null
$ echo $?
0
```

abellotti commented:
I saw the +90 −0 in the PR and it felt high.  No need to change, but PTAL at  https://github.com/quipucords/quipucords-installer/compare/2db827e...37919bd if you see anything you like. I tried to streamline the code a bit. Thanks.

ruda commented:
@infinitewarp wrote...

> The title of the original feature request was Ensure appropriate ownership and permissions of required host directories with this acceptance criteria...

I've updated this MR to verify ownership and permissions by the user. The output when the command `check` catches an error is this:

```
.../.local/share/quipucords/certs/server.key: ERROR: Incorrect permission(s): 0 ----------
.../.config/quipucords/env/env-ansible.env: ERROR: Incorrect permission(s): 444 -r--r--r--
.../.config/containers/systemd/quipucords-app.container: ERROR: Not owned by you (incorrect ownership): 0 root
```

I've changed the output when the file/directory is `OK`, it shows the permission and ownership. 

```
.../.local/share/quipucords/sshkeys: OK: drwxr-xr-x rmoura
```

**Question:  Is this too much verbose? Should we just say "OK"?**

ruda commented:
@infinitewarp wrote...

> When I run the new check command and search for those paths in the output, they appear to be "OK" when that is incorrect, and the script also exits with a successful 0 exit code which is also incorrect

The result codes are now consistent. In these examples, 9 errors were found.

Redirect stdout+stderr to /dev/null.

```
❯ ./bin/quipucords-installer check &> /dev/null
❯ echo $?
9
```

Redirect to `pipe grep` results the exit code of grep (zero).

```
❯ ./bin/quipucords-installer check | grep ERROR
/home/rmoura/.local/share/quipucords/certs/server.key: ERROR: Incorrect permission(s): 0 ----------
/home/rmoura/.local/share/quipucords/data/secret.txt: ERROR: Missing
/home/rmoura/.local/share/quipucords/db/userdata: ERROR: Incorrect permission(s): 0 d---------
/home/rmoura/.config/quipucords/env/env-ansible.env: ERROR: Incorrect permission(s): 444 -r--r--r--
/home/rmoura/.config/quipucords/env/env-app.env: ERROR: Incorrect permission(s): 444 -r--r--r--
/home/rmoura/.config/quipucords/env/env-celery-worker.env: ERROR: Incorrect permission(s): 444 -r--r--r--
/home/rmoura/.config/quipucords/env/env-db.env: ERROR: Incorrect permission(s): 444 -r--r--r--
/home/rmoura/.config/quipucords/env/env-redis.env: ERROR: Incorrect permission(s): 444 -r--r--r--
/home/rmoura/.config/quipucords/env/env-server.env: ERROR: Incorrect permission(s): 444 -r--r--r--
❯ echo $?
0
```

Found 9 errors, then the exit code is 9.

```
❯ ./bin/quipucords-installer check 
Checking Quipucords setup and configurations...
/home/rmoura/.local/share/quipucords: OK: drwxr-xr-x rmoura
/home/rmoura/.local/share/quipucords/certs: OK: drwxr-xr-x rmoura
/home/rmoura/.local/share/quipucords/data: OK: drwxr-xr-x rmoura
/home/rmoura/.local/share/quipucords/db: OK: drwxr-xr-x rmoura
/home/rmoura/.local/share/quipucords/log: OK: drwxr-xr-x rmoura
/home/rmoura/.local/share/quipucords/sshkeys: OK: drwxr-xr-x rmoura
/home/rmoura/.local/share/quipucords/certs/server.key: ERROR: Incorrect permission(s): 0 ----------
/home/rmoura/.local/share/quipucords/certs/server.crt: OK: -rw-r--r-- rmoura
/home/rmoura/.local/share/quipucords/data/secret.txt: ERROR: Missing
/home/rmoura/.local/share/quipucords/db/userdata: ERROR: Incorrect permission(s): 0 d---------
/home/rmoura/.config/quipucords: OK: drwxr-xr-x rmoura
/home/rmoura/.config/quipucords/env: OK: drwxr-xr-x rmoura
/home/rmoura/.config/quipucords/env/env-ansible.env: ERROR: Incorrect permission(s): 444 -r--r--r--
/home/rmoura/.config/quipucords/env/env-app.env: ERROR: Incorrect permission(s): 444 -r--r--r--
/home/rmoura/.config/quipucords/env/env-celery-worker.env: ERROR: Incorrect permission(s): 444 -r--r--r--
/home/rmoura/.config/quipucords/env/env-db.env: ERROR: Incorrect permission(s): 444 -r--r--r--
/home/rmoura/.config/quipucords/env/env-redis.env: ERROR: Incorrect permission(s): 444 -r--r--r--
/home/rmoura/.config/quipucords/env/env-server.env: ERROR: Incorrect permission(s): 444 -r--r--r--
/home/rmoura/.config/containers/systemd: OK: drwxr-xr-x rmoura
/home/rmoura/.config/containers/systemd/quipucords-app.container: OK: -rw-r--r-- rmoura
/home/rmoura/.config/containers/systemd/quipucords-celery-worker.container: OK: -rw-r--r-- rmoura
/home/rmoura/.config/containers/systemd/quipucords-db.container: OK: -rw-r--r-- rmoura
/home/rmoura/.config/containers/systemd/quipucords-redis.container: OK: -rw-r--r-- rmoura
/home/rmoura/.config/containers/systemd/quipucords-server.container: OK: -rw-r--r-- rmoura
/home/rmoura/.config/containers/systemd/quipucords.network: OK: -rw-r--r-- rmoura
❯ echo $?
9
```
